### PR TITLE
Added a config provider

### PR DESCRIPTION
**Pull request by Lefka**
Simple change to make the `uiMaskConfig` value into a provider. This just allows easy global configuration without having to edit the source files.

### CODE REVIEW

**Comment by NickTomlin** [comment]
This is a great idea! 

How is this going to affect existing users who are expecting a `value`? As long as we make this a major version bump I think it's okay to break them. Otherwise perhaps the provider should modify a shared value for backwards compatibility?

File: test/maskSpec.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ describe("uiMask", function () {
       expect(scope.x).toBe("abc123");
     });
 
-	it("should not dirty or invalidate the input", function() {
+    it("should not dirty or invalidate the input", function() {
```
Comment (NickTomlin) [inline test/maskSpec.js]: Looks like this is just a bunch of tabs => spaces. Can we keep this out of the PR and make that change in another?

**Comment by lukepfeiffer10** [comment]
Thanks for doing this! I was always thinking I should make a PR for this but never got around to doing it.

Also, as @NickTomlin suggested would this break a use case such as:

```javascript
app.run([ 'uiMaskConfig',
        function (uiMaskConfig) {
            uiMaskConfig.clearOnBlur = false;
            var isAndroid = /(android)/i.test(navigator.userAgent);
            if (isAndroid) {
                var index = uiMaskConfig.eventsToHandle.indexOf('input');
                uiMaskConfig.eventsToHandle.splice(index, 1);
            }
        }
    ]);
```

**Comment by Lefka** [comment]
You are both right. The scenario mentioned would have broken with the change as it was written. I think it's worth including the value as it was, if nothing else, for backwards compatibility. I updated the provider to merge the value and the provider config on get. This should allow the scenario to work fine.

I think this might need some unit tests now so let me look at that before this gets merged.

**Comment by Lefka** [comment]
Unit tests are in. Let me know if any other changes are needed.

**Pull request merged**